pkg/scheduler/eventhandlers.go

```diff
@@ -527,6 +532,26 @@ func addAllEventHandlers(
 	return nil
 }
 
+// podSchedulingPropertiesChange interprets the update of a pod and returns corresponding UpdatePodXYZ event(s).
+func podSchedulingPropertiesChange(newPod *v1.Pod, oldPod *v1.Pod) []framework.ClusterEvent {
+	var events []framework.ClusterEvent
+
+	if labelsChanged(newPod.ObjectMeta, oldPod.ObjectMeta) {
+		events = append(events, queue.PodLabelChange)
+	}
+
+	if podResourceRequestChanged(newPod, oldPod) {
+		events = append(events, queue.PodRequestChange)
+	}
+
+	if len(events) == 0 {
```

Comment:
Shouldn't we always consider AssignedPodUpdate? Otherwise we might break out-of-tree plugins.

Reply:
Yes,

> the update will be interpreted as PodLabelChange, and this PodLabelChange will be matched with Pod/Update as well = QHint registered for Pod/Update will be called with PodLabelChange event. 
https://github.com/kubernetes/kubernetes/pull/122628#discussion_r1451646204.

Reply:
Updated.